packages/react-components/src/organisms/Form.tsx

```diff
@@ -53,11 +51,12 @@ const Form = <T extends void | Record<string, unknown>>({
     if (formRef.current!.reportValidity() && parentValidation) {
       setStatus('isSaving');
       try {
-        const result = await onSave();
-        if (formRef.current) {
-          setStatus('hasSaved');
-        }
-        return result;
+        return await onSave().then((result) => {
+          if (formRef.current) {
+            setStatus('hasSaved');
+          }
+          return result;
+        });
```

Comment:
Mixing try...catch with a promise chain

Reply:
This was driving me crazy. Without it it seems to set status after the result has been returned so i get the batcher error. I might convert it all to be a promise chain though.

Reply:
I have discussed this with @markcsims as it seems like a weird race condition forcing me to do this in the admittedly weird way. He is going to give it a go and see if I am missing something as whats going on doesn't make sense to me.